examples/js/plugins/COGParser.js

```diff
@@ -0,0 +1,306 @@
+/* global itowns, THREE */
+
+/**
+ * Select the best overview level (or the final image) to match the
+ * requested extent and pixel width and height.
+ *
+ * @param {Source} source The COGSource
+ * @param {Extent} requestExtent The window extent.
+ * @param {number} requestWidth The pixel width of the window.
+ * @param {number} requestHeight The pixel height of the window.
+ * @returns {Object} The selected zoom level.
+ */
+function selectLevel(source, requestExtent, requestWidth, requestHeight) {
+    // Number of images  = original + overviews if any
+    const cropped = requestExtent.clone().intersect(source.extent);
+    // Dimensions of the requested extent
+    const extentDimension = cropped.planarDimensions();
+
+    const targetResolution = Math.min(
+        extentDimension.x / requestWidth,
+        extentDimension.y / requestHeight,
+    );
+
+    let level;
+
+    // Select the image with the best resolution for our needs
+    for (let index = source.levels.length - 1; index >= 0; index--) {
+        level = source.levels[index];
+        const sourceResolution = Math.min(
+            source.dimensions.x / level.width,
+            source.dimensions.y / level.height,
+        );
+
+        if (targetResolution >= sourceResolution) {
+            break;
+        }
+    }
+
+    return level;
+}
+
+/**
+ * Returns a window in the image's coordinates that matches the requested extent.
+ *
+ * @param {Source} source The COGSource
```

Comment:
Add implicit type parameters for correct typechecking:
```suggestion
 * @param {Object} source The COGSource
 * @param {number[]} source.origin Root image origin as an XYZ-vector
```

Reply:
Ok, done !